### PR TITLE
Complete merchant order fulfillment functionality

### PR DESCRIPTION
**Pull request by johnktravers**

### CODE REVIEW

File: app/controllers/merchant/orders_controller.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    item_order.update(status: 1)
+
+    item = item_order.item
+    item.reduce_inventory(item_order.quantity)
+    item.save
+
+    flash[:success] = ["You have successfully fulfilled #{item.name} for Order ##{item_order.order_id}"]
+    redirect_to "/merchant/orders/#{item_order.order_id}"
+  end
+end
```
Comment (zacisaacson) [inline app/controllers/merchant/orders_controller.rb:23]: Looks great!

File: app/models/item_order.rb [hunk 2]
```diff
@@ -4,7 +4,13 @@ class ItemOrder <ApplicationRecord
   belongs_to :item
   belongs_to :order
 
+  enum status: %w(unfulfilled fulfilled cancelled)
```
Comment (zacisaacson) [inline app/models/item_order.rb:7]: enum, nice!

File: app/views/merchant/dashboard/index.html.erb [hunk 3]
```diff
@@ -1,34 +1,4 @@
...
-        <td class='total-quantity'><%= @merchant.total_items_in_order(order) %></td>
-        <td><%= number_to_currency(@merchant.total_value_in_order(order)) %></td>
-      </tr>
-    <% end %>
-  </tbody>
-</table>
+<%= render partial: 'partials/merchant_dashboard', locals:
+  {
+    merchant_items_path: '/merchant/items'
+  } %>
```
Comment (zacisaacson) [inline app/views/merchant/dashboard/index.html.erb:4]: I like the render!

File: app/controllers/user_orders_controller.rb [hunk 4]
```diff
@@ -13,7 +13,12 @@ def update
     order = Order.find(params[:id])
     order.update(status: 3)
     order.item_orders.each do |item_order|
-      item_order.update(status: 'cancelled')
+      if item_order.fulfilled?
+        item = item_order.item
+        item.increase_inventory(item_order.quantity)
```
Comment (grwthomps) [inline app/controllers/user_orders_controller.rb:18]: Nice use of helper method

File: app/views/merchant/orders/show.html.erb [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+              <p class="inactive">Cancelled</p>
+            <% end %>
+          </td>
+        </section>
+      </tr>
+    <% end %>
+  </tbody>
+</table>
+
+<br>
```
Comment (zacisaacson) [inline app/views/merchant/orders/show.html.erb:56]: Great styling!

File: app/models/item_order.rb (hunk 2 above)
Comment (grwthomps) [inline app/models/item_order.rb:7]: I like the change to enum here

File: app/views/admin/dashboard/merchant_index.html.erb [hunk 6]
```diff
@@ -0,0 +1,4 @@
+<%= render partial: 'partials/merchant_dashboard', locals:
+  {
+    merchant_items_path: "/merchants/#{@merchant.id}/items"
+  } %>
```
Comment (grwthomps) [inline app/views/admin/dashboard/merchant_index.html.erb:4]: Nice job with using partials

File: spec/features/default_user/orders_index_spec.rb [hunk 7]
```diff
@@ -25,7 +25,7 @@
     visit '/profile/orders'
 
     within "#order-#{@order_1.id}" do
-      expect(page).to have_link(@order_1.id)
```
Comment (zacisaacson) [inline spec/features/default_user/orders_index_spec.rb:28]: Yes! Thank you for doing this!

**Review by grwthomps — APPROVED** [review]

**Review by zacisaacson — APPROVED** [review]

**Pull request merged**